kernel/Utilities/GlobalIndexing.h

```diff
@@ -493,6 +423,78 @@ class GlobalIndexing {
                    globalIndex < blockStart_ + numberOfBasisFunctionsInBlock_;
         }
     };
+
+    /**
+     * Visitor to access the *Offsets_ members (e.g. elementOffsets_) of the
+     * Offsets class. If there is no offset entry available for the MeshEntity
+     * then it will be created (and initialized to the default 0).
+     */
+    struct OffsetVisitor : public Base::MeshEntityVisitor<> {
+        OffsetVisitor(Offsets& offsets)
+            : visitOffsets_(offsets), entityOffset_(nullptr){};
+
+        void visit(Base::Element& element) final {
+            entityOffset_ = &visitOffsets_.elementOffsets_[element.getID()];
+        }
+        void visit(Base::Face& face) final {
+            entityOffset_ = &visitOffsets_.faceOffsets_[face.getID()];
+        }
+        void visit(Base::Edge& edge) final {
+            entityOffset_ = &visitOffsets_.edgeOffsets_[edge.getID()];
+        }
+        void visit(Base::Node& node) final {
+            entityOffset_ = &visitOffsets_.nodeOffsets_[node.getID()];
+        }
+
+        Offsets& visitOffsets_;
+        /// Pointer to the entry, possibly newly initialized
+        int* entityOffset_;
```

Comment:
why a pointer, does it have to be `nullptr` why not a copy? Just asking. :)

Reply:
Looking at the `ConstOffsetVisitor` I might not understand this class

Reply:
The intention is that it allows editing the offset. So this is either
 - nullptr before the visit
 - a pointer to an actual offset after the visit.
Note though that it can't be a reference, as that could only be initialized in the constructor, while we only have the information after visiting.

Reply:
Okay

Reply:
Are we on c++17 yet ? Std::optional

Reply:
> Are we on c++17 yet ?

Not yet, though looking at the gcc version on the cluster I think we can switch over at some point